cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -380,6 +385,35 @@ macro(ocv_check_compiler_optimization OPT)
   endif()
 endmacro()
 
+macro(ocv_check_compiler_optimization_runtime OPT)
+  if(NOT DEFINED CPU_${OPT}_RUNTIME_SUPPORTED)
+    if(CPU_${OPT}_SUPPORTED AND CPU_${OPT}_RUNTIME_TESTS)
+      if(__is_from_baseline)
+        if(NOT "x${CPU_${OPT}_FLAGS_ON}" STREQUAL "x")
+          set(_compile_flags "${CPU_${OPT}_FLAGS_ON}")
+        else()
+          set(_compile_flags "${CPU_BASELINE_FLAGS}")
+          ocv_append_optimization_flag(_compile_flags ${OPT})
+        endif()
+        foreach(_testname ${CPU_${OPT}_RUNTIME_TESTS})
+          set(_var_runname "HAVE_CPU_${OPT}_RUNTIME_${_testname}")
+          if(CPU_${OPT}_${_testname}_TEST_FILE)
```

Comment:
This looks like over complicated.
Lets avoid modification of this file - it is primary used to support compilation modes with dispatched code.

It is better to move compiler "workarounds" into `OpenCVCompilerOptions.cmake`
- after `ocv_compiler_optimization_options_finalize()` call
- check is applied with "baseline" compilation flags only (like current implementation)

In `OpenCVCompilerOptions.cmake` add something like this:
```
if((PPC64LE AND NOT CMAKE_CROSSCOMPILING) OR OPENCV_FORCE_COMPILER_CHECK_VSX_ALIGNED)
  ocv_check_runtime_flag("" "OPENCV_CHECK_VSX_ALIGNED" "${OpenCV_SOURCE_DIR}/cmake/checks/runtime/cpu_vsx_aligned.cpp")
endif()
if(OPENCV_CHECK_VSX_ALIGNED)
  set(OPENCV_CPU_RUNTIME_DEFINITIONS_CONFIGMAKE "${OPENCV_CPU_RUNTIME_DEFINITIONS_CONFIGMAKE}\n
#define CV_CPU_RUNTIME_VSX_ALIGNED
")
endif()
```


Also rename `CV_CPU_RUNTIME_VSX_ALIGNED` => `CV_COMPILER_VSX_ALIGNED`:
- don't forbid cross-compilation (remove "RUNTIME")
- declare this as a "compiler" workaround (instead of CPU)

Reply:
I thought it could be useful for other platforms, however I agree with you, your suggestion is more reasonable and there's no need to make it over complicated, thank you.